cmd/humanInsertion/humanInsertion.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/maf"
+	"log"
+)
+
+func humanInsertion(mafFile string, outBed_ins string, outBed_del string, species_ins string, species_del string) {
+	//initialize variables
+	mafRecords := maf.Read(mafFile) //Read entire mafFile. mafRecords has type Maf
+	var bedList_ins []*bed.Bed      //initialize 2 bed files
+	var bedList_del []*bed.Bed      //1 bed file for ins, 1 bed file for del
+
+	//go through each line
+	for i, _ := range mafRecords { //each i is a block
+		for k, _ := range mafRecords[i].Species { //each k is a line
+
+			//convert maf to bed, start with getting assembly because it is needed to verify species_ins and species_del
+			assembly_del, chrom_del := maf.SrcToAssemblyAndChrom(mafRecords[i].Species[k].Src)   //start with species_del, get assembly (e.g. panTro6), chrom (e.g. chrI)
+			assembly_ins, chrom_ins := maf.SrcToAssemblyAndChrom(mafRecords[i].Species[k-1].Src) //then find corresponding species_ins line, same block, 1 line above since pairwise maf
```

Comment:
When I tried to run humanInsertion.go I got an index out of range error that seems to be related to this line and line 79. Does Species[k-1] not work? Why not?

go run humanInsertion/humanInsertion.go ../../LoweLab/Results/hg38.panTro6.synNet_head.maf ../../LoweLab/Results/outBed_hg38.bed ../../LoweLab/Results/outBed_panTro6.bed hg38 panTro6

panic: runtime error: index out of range

goroutine 1 [running]:
main.humanInsertion(0x7fffc967a18e, 0x32, 0x7fffc967a1c1, 0x25, 0x7fffc967a1e7, 0x28, 0x7fffc967a210, 0x4, 0x7fffc967a215, 0x7)
        /mnt/c/Users/raven/Desktop/gonomics/cmd/humanInsertion/humanInsertion.go:23 +0x973
main.main()
        /mnt/c/Users/raven/Desktop/gonomics/cmd/humanInsertion/humanInsertion.go:79 +0xf4
exit status 2

Reply:
Yes, I think the k-1 is triggering the error.  Index out of range means you are trying to access a position in the slice (like arrays in many languages) that does not exist.  This can either be asking for a position with a negative index, or one beyond the last element of the slice.  You have two places where you use an index on that line and it could be the "i" or the "k-1".  I get used the line before and doesn't give an error, so k-1 looks like it could be it.
The for loop on line 19 sends k over all indices for the species in the maf block.  In this case we will generally have hg38 and panTro6 in the block, so k will first be 0 and then 1.  The first time through the loop k will be zero, or look at the first species in the block (probably hg38), so that makes k-1 equal to negative one, which is not allowed.
To verify this, I added the line:
log.Printf("score:%f k:%d k-1:%d\n", mafRecords[i].Score, k, k-1)
as line 20.
We can talk through this more at code review.  How you want to code this may depend on if you want to make it specific to pairwise mafs, where you could might be able to remove your 'k' loop and use hardcoded values of 0 and 1.  Otherwise I think you will have to use your k loop to find the index value of species you are looking for.